catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -577,18 +634,35 @@ protected ResourceInfo getResourceInfo(
         String metacardId = (String) value;
         LOGGER.debug("metacardId = {},   site = {}", metacardId, site);
         query = createMetacardIdQuery(metacardId);
-      }
+      } else if (ResourceRequest.GET_RESOURCE_BY_METACARD.equals(name)) {
+        // no need to actually perform the query, just create a resource info with the metacard
```

Comment:
❗ I'm not sure that this is exactly true. You may not need to perform the query in order to do this operation, but we count on this rolling through all of the security plugins around query in order to restrict who can view what metacards. Doing this, you're bypassing all of that logic and allowing someone to go straight to a resource. They could even brute force their way to any resource the system knows about.

Reply:
You have two options, as I see it. You can manually roll through the plugins here (not as desirable imo), or you can add implementations to the resource sections of the policy and access plugins (better) so we don't need to worry about this in the future, and you can continue to skip the query at this point.

Reply:
but that metacard would have been retrieved via query to start with right? This means it would have gone through all those plugins already ... unless I am missing something

Reply:
I do not have a public example but there are downstream examples of redaction plugins that redact all values except ID, source, and security attributes so you can understand why it was redacted.

Reply:
This is the security plugin Scott is referring to.  If we properly implemented this method then we would not have to rely on the query security to protect the resource.
https://github.com/codice/ddf/blob/0d9f282/catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java#L282-L285

Reply:
Also the policy plugins (but these should be pretty simple), for example: https://github.com/codice/ddf/blob/master/catalog/security/catalog-security-metacardattributeplugin/src/main/java/org/codice/ddf/catalog/security/policy/metacard/MetacardAttributeSecurityPolicyPlugin.java

Reply:
Removed this part of the PR and kept it around to re-introduce if need be later